src/syntax_analysis.rs

```diff
@@ -0,0 +1,192 @@
+use std::collections::HashMap;
+use syntect::parsing::syntax_definition::{
+    ContextReference, MatchOperation, MatchPattern, Pattern, SyntaxDefinition,
+};
+use syntect::parsing::{Scope, SyntaxSet, SyntaxSetBuilder};
+
+/// Used to look up (by name) what dependencies a given [SyntaxDefinition] has
+type SyntaxToDependencies = HashMap<String, Vec<Dependency>>;
```

Comment:
I'm okay with the doc comment too. Another route that I also like is to use typedefs to express the intent:
```
type SyntaxName = String;
type SyntaxToDependencies = HashMap<SyntaxName, Vec<Dependency>>;
```
which might not need a doc comment at all.